src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -234,7 +233,7 @@ bool FollowMovementGenerator<T>::PositionOkay(T* owner, Unit* target, float rang
     if (owner->GetExactDistSq(target) > G3D::square(owner->GetCombatReach() + target->GetCombatReach() + range))
         return false;
 
-    return !owner->IsPet() || !angle || angle->IsAngleOkay(target->GetRelativeAngle(owner)); // need to check - dont think we need !pet exception here because there are scripts with MoveFollow that require angle
+    return !angle || angle->IsAngleOkay(target->GetRelativeAngle(owner));
```

Comment:
we added that `!owner->IsPet()` check to fix something about pet. I don't remember exactly what but why you removed it? what was the bug?

Reply:
i dont know why you added pet check there. pet has to have right angle relative to owner as well. and requires this check